src/app.rs

```diff
@@ -295,13 +364,13 @@ fn autoclick(
                 ));
             }
 
-            // TODO: Implement non blocking smooth mouse movement (i.e. it should not mouse directly)
             // Move mouse to saved coordinates if requested
             if click_position == ClickPosition::Coord {
+                // move to final destination
```

Comment:
Can `move_to()` be implemented here instead?
I think we may remove `is_executing` and go back to `is_autoclicking` this way.
It will also solve the autoclick issue I mentioned below (the one I commented out)

The `send()` below on L370-373 could also be removed then